website/src/download.md

```diff
@@ -17,3 +17,45 @@
   ~ under the License.
 -->
 
+# Apache Iceberg-Rust™ Downloads
+
+The official Apache Iceberg-Rust releases are provided as source artifacts.
+
+## Releases
+
+The latest source release is [0.2.0](https://archive.apache.org/dist/iceberg/iceberg-rust-0.2.0/apache-iceberg-rust-0.2.0-src.tar.gz) ([asc](https://archive.apache.org/dist/iceberg/iceberg-rust-0.2.0/apache-iceberg-rust-0.2.0-src.tar.gz.asc),
```

Comment:
We must use link that point to `https://www.apache.org/dyn/closer.lua/iceberg/iceberg-rust-0.2.0/apache-iceberg-rust-0.2.0-src.tar.gz?action=download`.